src/main/java/tasks/Deadlines.java

```diff
@@ -0,0 +1,50 @@
+package tasks;
+import enums.DialogueTypes;
+import errors.InvalidDeadlineException;
+import managers.OutputDialogueManager;
+
+public class Deadlines extends Task{
+    public static final int DEADLINE_DIVIDER_LENGTH = 4;
+    private String itemName;
+    private String deadline;
+
+    public Deadlines(String itemName) throws InvalidDeadlineException {
+        super(itemName);
+        int indexOfDivider = itemName.indexOf("/by");
```

Comment:
You might want to make `"/by"` into a named constant to prevent the magic string issue. Same applies to other similar strings in other files. Look out for any posible magic strings/literals/numbers in your code.